crates/pathfinder/src/storage.rs

```diff
@@ -49,7 +50,7 @@ pub struct Storage(Inner);
 
 #[derive(Clone)]
 struct Inner {
-    database_path: PathBuf,
+    database_path: Arc<PathBuf>,
```

Comment:
`PathBuf::clone()` is a deep-clone, hence the `Arc`.

Reply:
I would add this as a comment then :D

Reply:
Hmm. Is there not some better container for this, since its effectively static for the lifetime of a storage.

Reply:
I don't like it either, lemme see..

Reply:
The problem here is that we can trade deep cloning for shallow cloning at the expense of ref counting. However ref counts need to be incremented/decremented in a thread safe manner, which means arc is the fastest solution here. Fastest as in:
- just works, no additional custom implementation needed,
- still faster performance wise than deep-cloning (allocating on the heap).

Reply:
Yeah that makes sense. And really, the cost is trivial compared to everything else so lets move on :D